lib/Net/Amazon/Signature/V4.pm

```diff
@@ -174,6 +174,16 @@ Maintained by Dan Book, C<< <dbook at cpan.org> >>
 
 =cut
 
+# https://github.com/aws/aws-sdk-java-v2/blob/a9387dcb384ef8d62e022483d7a10635f891d618/utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java#L41
+
+sub _aws_uri_escape {
+	my $elem = uri_escape(shift);
+	$elem =~ s/\+/%20/g;
```

Comment:
Encoding + and * to their percent equivalents and not encoding ~ is the default behavior of uri_escape, so the following "fixups" shouldn't be needed here.

Reply:
Probably was overzealous.